R/teal_data.R

```diff
@@ -29,11 +28,21 @@
 teal_data <- function(...,
                       join_keys = teal.data::join_keys(),
                       code = character(0),
-                      check = FALSE) {
+                      check) {
   data_objects <- rlang::list2(...)
   if (inherits(join_keys, "join_key_set")) {
     join_keys <- teal.data::join_keys(join_keys)
   }
+  if (!missing(check)) {
```

Comment:
How about using the recommended way of deprecating an argument (changing default to `check = deprecated()`)?

![image](https://github.com/insightsengineering/teal.data/assets/211358/028957df-fa36-4c36-910f-4581ed1a9d64)

![image](https://github.com/insightsengineering/teal.data/assets/211358/8c67a65a-7a8d-403b-a4fd-79164f38fb41)